cypress/support/commands.js

```diff
@@ -43,10 +44,42 @@ Cypress.Commands.overwrite('login', (login, user) => {
 	login(user)
 })
 
+Cypress.Commands.overwrite('visit', (originalFn, url, options) => {
+	// Make sure that each visit call that triggers a page load will update the stored requesttoken
+	return originalFn(url, options).then((result) => {
+		cy.window()
+			.then((win) => cy.wrap(win?.OC?.requestToken))
+			.as('requesttoken')
+	})
+})
+
+Cypress.Commands.add('getRequestToken', () => {
+	cy.get('@requesttoken')
+		.if((token) => !token)
+		.then(() => {
+			cy.window()
```

Comment:
Thanks for diving into this. Migth also prove helpful when using the `@nextcloud/cypress` login function in collectives.

I wonder if we could upstream this as it seems like a good addition in general.

Would be nice if we could get rid of the `cy.window()` call as that will only work after a `cy.visit()` if i understand correctly. Otherwise we could even use this function to get request tokens for requests without a loaded page.

Reply:
Yeah, i've found it quite hard to force cypress to do those conditionals even with cypress-if. Another downside of the alias approach is that this is not kept across beforeEach and actual tests.